Mathlib/Algebra/Algebra/Defs.lean

```diff
@@ -311,6 +324,45 @@ section
 
 end
 
+section compHom
+variable {R : Type u} {S : Type v} (A : Type w)
+    [CommSemiring R] [Semiring A] [Algebra R A] [CommSemiring S]
+    (f : S →+* R)
+
+/--
+Compose a Algebra with a RingHom, with action f s • m.
+-/
+abbrev compHom: Algebra S A where
+  smul s a := f s • a
+  toRingHom := (algebraMap R A).comp f
+  commutes' _ _ := Algebra.commutes _ _
+  smul_def' _ _ := Algebra.smul_def _ _
+
+theorem compHom_smul_def (s : S) (x : A):
+    letI := compHom A f
+    s • x = f s • x := by
+      letI := compHom A f
+      rw [Algebra.smul_def]
+      exact Eq.symm (smul_def (f s) x)
+
+theorem compHom_algebraMap_eq :
+    letI := compHom A f
+    algebraMap S A = (algebraMap R A).comp f := rfl
+
+theorem compHom_algebraMap_apply (s : S) :
+    letI := compHom A f
+    algebraMap S A s = (algebraMap R A) (f s) := rfl
+
+theorem compHom_toRingHom_eq :
+    letI := compHom A f
+    (compHom A f).toRingHom = (algebraMap R A).comp f := rfl
+
+theorem compHom_toRingHom_apply (s : S) :
+    letI := compHom A f
+    (compHom A f).toRingHom s = (algebraMap R A) (f s) := rfl
```

Comment:
My inclination would be to drop these lemmas; I think since you're using `abbrev` above, lean will unfold them with `dsimp only`.